### ISSUE TITLE
Error: Can't build `plugin/preact-rectangles` or `plugin/preact-resizable` templates

### ISSUE THREAD

**Issue by jacktowery** [yuanqing/create-figma-plugin#226]
I'm getting a TypeScript error when trying to build the `plugin/preact-rectangles` or `plugin/preact-resizable` templates for the first time:

```
$ npm run build

> build
> build-figma-plugin --typecheck --minify

info Typechecking...
error TypeScript error

    node_modules/@create-figma-plugin/ui/lib/types/event-handler.d.ts:3:86 - error TS2344: Type 'Parameters<NonNullable<EventHandler>>' does not satisfy the constraint 'never[]'.
      Type 'any[]' is not assignable to type 'never[]'.
        Type 'any' is not assignable to type 'never'.

    3 type FirstArgument<EventHandler extends ((...args: any) => any) | undefined> = First<Parameters<NonNullable<EventHandler>>>;
```

Steps to reproduce (following the [Quick Start](https://yuanqing.github.io/create-figma-plugin/quick-start/) guide):
1. `$ npx --yes create-figma-plugin`
2. Select `plugin/preact-rectangles` or `plugin/preact-resizable`
3. `$ cd preact-rectangles`
4. `$ npm run build`
5. TypeScript error

The same error occurs when attempting to add `@create-figma-plugin/ui` to an existing plugin following the [instructions on the documentation website](https://yuanqing.github.io/create-figma-plugin/ui/).

**Comment by artemartemov** [comment]
same issue here

**Comment by thevarunkhatri** [comment]
Having the same issue. Did some tinkering and it appears the problem is Typescript 5.4. Installing typescript 5.3 instead fixed it for me.

**Comment by plcdnl** [comment]
Same error https://github.com/yuanqing/create-figma-plugin/issues/227

**Comment by darkaico** [comment]
I ended up skipping the lib check to avoid this error, in `tsconfig.json` adding `skipLibCheck: true`

```
"compilerOptions": {
    ...
    "skipLibCheck": true
  },
...
}
```

**Comment by yuanqing** [comment]
This issue should be fixed as of `3.2.0`

```sh
$ npx --yes -- create-figma-plugin@3.2.0 --template plugin/preact-rectangles
info Copying "plugin/preact-rectangles" template...
info Resolving package versions...
info Installing dependencies...
success Done

First:
  cd preact-rectangles

To build the plugin:
  npm run build

To watch for code changes and rebuild the plugin automatically:
  npm run watch

$ cd preact-rectangles
$ npm run build

> build
> build-figma-plugin --typecheck --minify

info Typechecking...
success Typechecked in 0.850s
info Building...
success Built in 0.327s
```
